### ISSUE TITLE
Discussion Forum

### ISSUE THREAD

**Issue by dekyfin** [mailcow/mailcow-dockerized#2392]
Mailcow is such an awesome product. The developers and contributors have done such a great work. I realised that Mailcow doesn't have a discussion forum so all questions have to be posted here. 

I think it would be great if we can have a forum. I am willing to set one up if need be. Let me know what your plans are

**Comment by andryyy** [comment]
Hi,

Thank you for your kind words!

I was thinking about a good old WoltLab outside my own infrastructure. 

I cannot stand Discourse, so this is not an option. :(

**Comment by dekyfin** [comment]
I've never used WoltLab so I don't know much about it. I was thinking of nodeBB or phpBB.

**Comment by ntimo** [comment]
I actually like discourse to be honest. But WoldLab is also very very nice. The only problem with WordLab is that its pretty expensive. But what ever kind of forum software gets used at the end of the day, a forum would defently be nice :)

**Comment by Braintelligence** [comment]
I recommend Flarum

**Comment by tobias992** [comment]
https://de.wordpress.org/plugins/asgaros-forum/

...for WordPress. There is a very powerful solution - and for free.

I have a small Webhosting Company in Germany and if you want i can host this for the Community.

**Comment by extremeshok** [comment]
why do we need this ? wasted effort, moderations, admin time etc. 

Github works and if you need ideas / feature requests, something along the lines of https://feathub.com or like what has been working, submitting PR and Issues/feature requests.

As too hosting, most of us can sponsor dedicated servers, i'm sure [servercow.de](https://servercow.de) will be the host for everything this project requires.

**Comment by PhoenixPeca** [comment]
I think having a forum is really great! It helps us clear out some github issue threads that arent really mailcow issues. Besides, it's nice to have a place to find or give help from or to another people, and of course a place to appreciate mailcow.

For the forum software, I personally like discourse. The mailing list and email replies integrates seamlessly with mailcow emails server suite. Anyhow, any forum software will do the trick.

**Comment by itsbhanusharma** [comment]
I'd like to recommend discourse
[discourse.org](https://discourse.org)
There are many obvious advantages when it comes to using discourse compared to the traditional forum software which is why a lot of open source projects are using it to manage their community.

**Comment by dekyfin** [comment]
Let's focus on implementation for now. The forum platform to use will become obvious as we continue.

For now I and @tobias992 are willing to host it for the community. I can setup phpBB or nodeBB on my VPS (with a couple of other sites). It's not much, but it can get the job done alright. Anyone willing to provide a hosting platform can tell us about the platform they are offering

@tobias992... I would like to know about the hosting environment you're providing. This will help us to determine the platform to use.

Finally we will need admins/moderators. Anyone willing to help in that regard?

**Comment by tobias992** [comment]
Hello,

my hosting environment is fully optimized to work with WordPress. I really can recommend the Asgaros-forum plugin. I use it for a lot of my customers.
I use KVM machines (Xeon CPU, DDR4 ECC RAM, SSD RAID 10). First i would host it at one of my shared Hosting Servers. Then we can look at traffic, memory and disk usage. Maybe later i can host the forum at his own KVM machine (if needed).

I can do the hosting (only WordPress) if  andryyy  accept it.

**Comment by heavygale** [comment]
I would vote for [Discourse](https://www.discourse.org/) or [Flarum](https://flarum.org/), too. I don't really like those old fashioned forum systems like the mentioned phpBB, nodeBB or Asgaros. And Flarum too only needs PHP and a database, schould be easy to host.

**Comment by j4ndev** [comment]
I am able to sponsor a Woltlab Forum License, if needed. It is also easy to host on a VPS with PHP and database.

**Comment by itsbhanusharma** [comment]
I'd be able to Install & set-up Discourse (with all the required bells & whistles)
Hosting it will be slightly expensive compared to traditional forum but it'll be worth it. 

I'll be more than happy to kitty for the hosting perpetually.

**Comment by hachre** [comment]
> I was thinking about a good old WoltLab outside my own infrastructure.
> 
> I cannot stand Discourse, so this is not an option. :(

Andryyy already stated his preferences, he's against Discourse but for WoltLab

**Comment by Braintelligence** [comment]
@extremeshok There are topics that don't belong here and get flooded by other content on IRC in time.
Content/questions, that is perhaps too shallow for the docs but is appearing here and stealing @andryyy precious time and energy by answering redundantly all the time.

**Comment by andryyy** [comment]
WoltLab would be nice indeed. :)

**Comment by dekyfin** [comment]
It stands to be that @andrrry's preference for WoltLab carries a lot of
weight and cannot be undermined.

Some of us would like to know why you favour WoltLab and can't stand
Discourse.

The major advantage Discourse has over other forums is the response by
email feature which GitHub has made us come to love

**Comment by andryyy** [comment]
It just sucks.

Seriously, it does not help if I start to explain, I will still not like it.

**Comment by dekyfin** [comment]
I think it's clear now: Discourse is out of questions. We'll have to go
with the WoltLab.

@janlsm, we'll be taking you up on your offer for the license. All that is
left is the hosting and a couple of moderators

**Comment by Braintelligence** [comment]
Discourse might be out of question, but I see nothing against Flarum.

**Comment by dekyfin** [comment]
Flarum is a good software. I've used it once. My main concern is that it's
still in beta

**Comment by Braintelligence** [comment]
It's in beta for a long time now, though. I know multiple Forums that use it in production without any serious issues up till now.

**Comment by cklabautermann** [comment]
Flarum is nice imho, too. Easy to administer and not feature-overloaded.

**Comment by j4ndev** [comment]
So, what's the plan now?

**Comment by dekyfin** [comment]
I've set up a demo for both flarum and woltlab. We need a few people to
test the two and give their feedback.

Woltlab demo:
https://39e46c.woltlab-demo.com/
Access credentials:
Username: demo
Password: uzl71tapsa

Admin Credentials:
Username: admin
Password: admin


Flarum demo:
https://mailcow.freeflarum.com <Https://mailcow.freeflarum.com>
login: admin
pass: ghqruUEQ

**Comment by dekyfin** [comment]
I've set up a demo for both flarum and woltlab. We need a few people to
test the two and give their feedback.

Woltlab demo:
https://39e46c.woltlab-demo.com/
Access credentials:
Username: demo
Password: uzl71tapsa

Admin Credentials:
Username: admin
Password: admin


Flarum demo:
https://mailcow.freeflarum.com 
login: admin
pass: ghqruUEQ

**Comment by hachre** [comment]
I've quickly looked at both, Woltlab looks more like a traditional forum software. Flarum looks more like a Discourse style software.

Depending on what Andryyy dislikes about Discourse, Flarum might be out of the question as well, because it has the same general feel to it. 

I am good either way!

**Comment by Braintelligence** [comment]
Just to be clear: The dev community of Flarum is not affiliated with Discourse. The main devs came from esoTalk: https://esotalk.org/

**Comment by andryyy** [comment]
Use whichever you like. 😄

I would love WoltLab, but it isn't free. Flarum looks like Discourse, but I'm fine with it, as long as you guys like it.

**Comment by extremeshok** [comment]
**https://mailcow.farm**

https://mailcow.farm/index.php?thread/1-welcome-to-the-mailcow-community-forum/#post1

I have set this up with the goal of advancing mailcow and to fulfill the wishes of the community. Actions speak louder than words.

Why mailcow.farm ? Simple.... cows live on farms ;P

This entire site is not-for-profit. There will be no advertising, adverts, user tracking, etc. I personally can not stand it and I have enough resources to fund this.

A little about me: github.com/extremeshok
Forum experience: ran a 10 000+ member gaming forum for a long time.

As the site grows, I will allocate more resources, having a virtually unlimited capacity of dedicated servers, vm;s etc.

Site is hosted in Germany and is GDPR compliant, daily backups (encrypted), dns is anycast.

Server is setup on an optimized version of ubuntu lts running docker-compose, no control panels, no ftp, no email servers, no password logins.

Full admin permissions will be given to andryyy (mailcow developer)

If requested all members of mailcow project (as per github) will be given moderator.

Forum software is a commercial (non-free/paid) version of wotlab, paid for by myself specifically for mailcow.farm.

I have attached a graphical representation of the topology of the docker-compose, most of the docker images used are on github.

![docker-vis-full](https://user-images.githubusercontent.com/5957328/54329665-465be680-461b-11e9-82b4-28f2b7b02fdc.png)
